test_horizons_client.py

```diff
@@ -0,0 +1,125 @@
+import pytest
+from hypothesis import given, strategies as st
+
+from horizons import Coordinates, Angle, horizons_session
+from common.minimodel import Site, NonsiderealTarget, TargetTag, TargetType
+from datetime import datetime
+
+@pytest.fixture
+def target():
+    return NonsiderealTarget('Jupiter', None, type=TargetType.BASE,
+                             tag=TargetTag.MAJOR_BODY, des='jupiter', ra=None, dec=None)
+
+@pytest.fixture
+def session_parameters():
+    return (Site.GS, datetime(2019, 2, 1), datetime(2019, 2, 1, 23, 59, 59), 300)
+
+
+@given(
+    st.lists(
+        st.floats(allow_infinity=False, allow_nan=False),
+        min_size=4,
+        max_size=4,
+    )
+)
+def test_angular_distace_between_values(values):
+    """
+    Angular Distance must be in [0, 180°]
+    """
+    a, b, c, d = values
+    assert Coordinates(a, b).angular_distance(Coordinates(c, d)) <= 180
+
+
+
+@given(
+    st.lists(
+        st.floats(allow_infinity=False, allow_nan=False),
+        min_size=2,
+        max_size=2,
+    )
+)
+def test_angular_distace_between_any_point_and_itself(values):
+    """
+    Angular Distance must be zero between any point and itself
+    """
+    a, b = values
+    assert Coordinates(a, b).angular_distance(Coordinates(a, b)) == 0
+
+@given(
+    st.lists(
+        st.floats(allow_infinity=False, allow_nan=False),
+        min_size=4,
+        max_size=4,
+    )
+)
+def test_angular_distace_symmetry(values):
+    """
+    Angular Distance must be symmetric to within 1µas
+    """
+    a, b, c, d = values
+    delta = Coordinates(a, b).angular_distance(Coordinates(c, d)) - Coordinates(c, d).angular_distance(Coordinates(a, b))
+    assert Angle.to_signed_microarcseconds(delta) <= 1
```

Comment:
I was thinking about the signed microarcseconds... I need to look a bit more closely, but I think that these tests will fail because the data we get from OCS is not as precise as the data we get from GPP, so converting TO signed microarcseconds from a float will not offer the same precision. We should probably be doing tests that are less precise in these cases? When you are back from vacation, let's discuss what is failing and why: either that, or I will look at it this week.